server/internal/auth/check.go

```diff
@@ -0,0 +1,37 @@
+package auth
+
+import (
+	"context"
+	"log/slog"
+	"net/http"
+	"server/internal/encrypt"
+)
+
+// Мидлварь, который проверяет аутентификацию текущего пользователя
+func (i *Item) Check(h http.Handler) http.Handler {
+	authFn := func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+
+		authLogin := r.Header.Get("Login")
+		authPass := r.Header.Get("Password")
+
+		model, err := i.storage.UserGet(ctx, authLogin)
+		if err != nil {
+			slog.Error("error when get user from storage", "err", err)
+			http.Error(w, "Internal Server Error", http.StatusInternalServerError)
+			return
+		}
+
+		i := encrypt.Item{}
+
+		if i.CheckPasswordHash(authPass, model.HashedPassword) {
+			ctx = context.WithValue(ctx, UserIDKey, model)
+		} else {
+			w.WriteHeader(http.StatusUnauthorized)
+			return
+		}
+
+		h.ServeHTTP(w, r.WithContext(ctx))
```

Comment:
Я бы советовал использовать `comma - ok` подход к проверке будлевых значений, который принят  в Go. По-моему это просто красиво 😊 
```suggestion
                if ok := i.CheckPasswordHash(authPass, model.HashedPassword); !ok {
			w.WriteHeader(http.StatusUnauthorized)
			return
		}

		ctx = context.WithValue(ctx, UserIDKey, model)
		h.ServeHTTP(w, r.WithContext(ctx))
```

Reply:
Переделал. Непривычно для меня)

Reply:
С утилитарной точки зрения я это себе объясняю, как возможность дать информативное имя локальной переменной добавив немного контекста тем самым.